Ft get all post from database

bizip opened the pull request:
#### What does this PR do?

- Retrieve all post form database

- Dispay a massage with in json format

#### Description of Task to be completed

All task described above are completed 

#### How should this be manually tested?
Step 1: Go to https://github.com/bizip/Pascal-capstone-project

Step 2: Click on code (Button)
![image](https://user-images.githubusercontent.com/52130946/93663232-8698f200-fa66-11ea-9222-e7517065af61.png)


Step 3: Copy string of code

Step 4: Open your terminal on your local machine and change directory to desktop

Step 5: type git clone and paste the link then press enter

Step 6: Type cd pascal-capstone-project

step 7: Run npm install and all package will be installed

Step 8: type npm start  

Step 9: open postman type URL provided in the screen shot provided below

![image](https://user-images.githubusercontent.com/52130946/93707102-397f5380-fb2c-11ea-942d-628e670c98b3.png)


#### What are the relevant Trello story?

https://trello.com/c/AVB7PC0c

bizip committed:
ft(All query/question): display  list of all questions

-Displayed all list of query/question on one page
-Created delete botton to delete single question
-Added a well displayed message when delete action is done

[ Done ]

bizip committed:
bg(fixed delete): fixed issue on delete button

[ done ]

bizip committed:
ft(Send Question): Send question in firebase

-Connected contact form to firebase
-Well displayed message when a message is successfull sent
-Reset form once action is done using promises

[ Done ]

bizip committed:
ft(Post article): Post new aricle in firebase

-Connected new article form to firebase
-Displayed a success message once aricle is successfully sent to firebase

[ Done ]

bizip committed:
ch(Startup files ): Setting up database connection

-Created startup folder structure as required
-Added configuration file like .env, .babelrc
-Created a connection to MongoDB

[ Done ]

bizip committed:
ft(Get post): Getting all post form database

-Retrieve all post form database
-Dispay a massage with in json format

[ Done ]

mukunzidd commented on .env:
```diff
@@ -0,0 +1,3 @@
+HOST=localhost
+PORT=8000
+DB_URL='mongodb+srv://bizimungu:0728038pp.PP@agakiriro.0xgn1.mongodb.net/blog?retryWrites=true'
```
This file should not be on GitHub. They are your app's secrets, exposing them gives anyone on the internet to get into your DB and  do whatever they want there.

bizip replied:
Done

mukunzidd commented on UI/html/blog.html:
```diff
@@ -75,6 +75,7 @@ <h1>Subscribe to for news letter</h1>
+        <p id="deleteArticle"></p>
         <div class="all__blog">
             <div class="item" id="single__item">
                 <!--  <div class="article" id="blog__list"> -->
```
Remove this line if you're not using it.

bizip replied:
Done

mukunzidd commented on UI/html/question.html:
```diff
@@ -75,32 +75,33 @@ <h1>Subscribe to for news letter</h1>
                 <!-- <div class="article"> -->
-                <h4><b>Benimana Eric</b></h4>
+                <!-- <h4><b>Benimana Eric</b></h4>
                 <p class="article__text">Prosciutto beef ribs strip steak, biltong chislic tri-tip shankle. Biltong drumstick alcatra, shankle leberkas brisket pork belly pork chop. Rump andouille pork chop, beef ribs fatback t-bone brisket shankle tur salami short ribs corned
                     beef tongue. <br>
                     <p id="like__icon"><b>bizip04@gmail.com</b>&nbsp;&nbsp;&nbsp;&nbsp;</p>
 
                     <p id="like__icon"><b>Delete</b></p>
 
```
Remove unused code.

bizip replied:
Done

mukunzidd commented on UI/html/question.html:
```diff
@@ -75,32 +75,33 @@ <h1>Subscribe to for news letter</h1>
+            <!-- <div class="item" id="single__article"> -->
+            <!-- <div class="article"> -->
+            <!-- <h4><b>Benimana Eric</b></h4>
                 <p class="article__text">Prosciutto beef ribs strip steak, biltong chislic tri-tip shankle. Biltong drumstick alcatra, shankle leberkas brisket pork belly pork chop. Rump andouille pork chop, beef ribs fatback t-bone brisket shankle tur salami short ribs corned
                     beef tongue. <br>
                     <p id="like__icon"><b>bizip04@gmail.com</b>&nbsp;&nbsp;&nbsp;</p>
                     <p id="like__icon"><b>Delete</b></p>
 
```
Remove this too. The reason you need to remove this is because a PR is an increment to the whole product and when you deliver a feature/story you have to deliver what was asked nothing more.

bizip replied:
Done

bizip replied:
Done

mukunzidd commented on UI/js/all__blog.js:
```diff
@@ -21,7 +21,17 @@ let renderBlog = (doc) => {
         e.stopPropagation();
         let id = e.target.parentElement.getAttribute('data-id');
-        // db.collection('articles').doc(id1).delete;
+
+        db.collection('articles').doc(id).delete().then(function() {
+            let deletDoc = document.getElementById("deleteArticle");
+            deletDoc.textContent = "Document successfully deleted!";
+            deletDoc.className = "alert__message";
+            setTimeout(() => {
+                window.location.reload();
+            }, 4000);
+        }).catch(function(error) {
+            console.error("Error removing document: ", error);
+        });;
```
This code should only be in a PR that handles deleting an article. @bizip

bizip replied:
I have moved this deleting functionality in this PR

bizip committed:
ch(Working on feedback): Correcting some mistakes

-Removed unused line of code
-Removed unnecessary commented code
-Moved deleting functionality from this PR to deleting PR

[ Done ]
